README.md

```diff
@@ -162,6 +168,139 @@ Click on the button below to deploy this demo to the IBM Cloud.
+APP_NAME                                      The name assigned to all of the frontend objects defined in this template.                                                    nlu-code-pattern
+NATURAL_LANGUAGE_UNDERSTANDING_AUTH_TYPE      How to authenticate with the NLU service, possible values are: iam, cp4d, bearerToken                                         bearerToken
+NATURAL_LANGUAGE_UNDERSTANDING_BEARER_URL     The service URL.
+NATURAL_LANGUAGE_UNDERSTANDING_BEARER_TOKEN   Cloud Pak for Data bearer token.
+NATURAL_LANGUAGE_UNDERSTANDING_APIKEY         IBM Cloud IAM apikey from the service instance page.
+NATURAL_LANGUAGE_UNDERSTANDING_USERNAME       Cloud Pak for Data username of user with access to the service instance.
+NATURAL_LANGUAGE_UNDERSTANDING_PASSWORD       Cloud Pak for Data password of user with access to the service instance.
+NAMESPACE                                     The OpenShift Namespace where the ImageStream resides.                                                                        openshift
+APPLICATION_DOMAIN                            The exposed hostname that will route to the Node.js service, if left blank a value will be defaulted.                         nlu-code-pattern
+SOURCE_REPOSITORY_URL                         The URL of the repository with your application source code.                                                                  https://github.com/watson-developer-cloud/natural-language-understanding-code-pattern
+SOURCE_REPOSITORY_REF                         Set this to a branch name, tag or other ref of your repository if you are not using the default branch.
+```
+
+Depending on where you service instance is running you may have to provide different parameters to this application. Follow the [instructions](#manually) on how to run the application locally and create the `.env` file with the service credentials. The same values will be used when deploying to openshift.
+
+Create the template with the values from the previous step.
+
+```
+oc new-app -f openshift/template.yaml --ignore-unknown-parameters=true --param-file ./env
+```
+
+Kick off a build once all required dependencies are confirmed
+
+```bash
+oc nlu-code-pattern
+```
+
+#### Check the status
+
+To check the status of the application run:
+
+```bash
+oc status
+```
+
+Which should return something like:
+
+```
+In project NLU Code Pattern Project (nlu-project) on server https://10.2.2.2:8443
+
+  svc/nlu-code-pattern - 172.30.108.183:8080
+  dc/nlu-code-pattern deploys istag/nlu-code-pattern:latest <-
+    bc/nlu-code-pattern source builds https://... on openshift/nodejs:12
+      build #1 running for 7 seconds
+    deployment #1 waiting on image or update
+```
+
+#### Custom Routing
+
+An OpenShift route exposes a service at a host name, like www.example.com, so that external clients can reach it by name.
```

Comment:
OpenShift is suddenly capitalized properly... I have a feeling you borrowed this snippet from somewhere 👀